### ISSUE TITLE
[LINT_BUG]: box_usage_linter false positive when mixing packages with scripts in box::use

### ISSUE THREAD

**Issue by TymekDev** [Appsilon/box.linters#150]
### box.linters version

v0.10.5

### Sample source code to lint

```r
box::use(
  app/main,
  shiny[shinyApp],
)


shinyApp(main$ui, main$server)
```

### Lint command used

```r
code <- "box::use(
  app/main,
  shiny[shinyApp],
)


shinyApp(main$ui, main$server)"

lintr::lint(
  code,
  linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters)
)
```

### Lint result

```
<text>:7:1: warning: [box_usage_linter] Function not imported nor defined.
shinyApp(main$ui, main$server)
^~~~~~~~
```

### Expected result

I expected to not receive a lint as `shinyApp` is imported.

**Comment by radbasa** [comment]
I'm surprised the `box_separate_calls_linter` did not throw lint.

https://appsilon.github.io/box.linters/reference/box_separate_calls_linter.html
